aerospike-core/src/cluster/mod.rs

```diff
@@ -51,8 +51,8 @@ pub struct Cluster {
     // Active nodes in cluster.
     nodes: Arc<RwLock<Vec<Arc<Node>>>>,
 
-    // Hints for best node for a partition
-    partition_write_map: Arc<RwLock<HashMap<String, Vec<Arc<Node>>>>>,
+    // Which partition contains the key.
+    partition_write_map: RwLock<HashMap<String, [Option<Arc<Node>>; node::PARTITIONS]>>,
```

Comment:
The Arc wasn't really serving any purpose. So I removed it... however now partitions are Options so we know if no node has claimed a partition and can bubble an error to the caller.